types/wlr_surface_layers.c

```diff
@@ -0,0 +1,129 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <wayland-server.h>
+#include <wlr/types/wlr_surface.h>
+#include <wlr/types/wlr_surface_layers.h>
+#include <wlr/util/log.h>
+#include "surface-layers-protocol.h"
+
+static const char *surface_layers_role = "layer_surface";
+
+// TODO
+#define SURFACE_LAYERS_ERROR_ROLE 1
```

Comment:
These are normally defined within the protocol definition.

Reply:
Yeah, exactly. Since it's not defined in the XML file yet, I wanted to talk with @SirCmpwn before adding it.

Reply:
Yeah stick that in there please.